spec/syntax.md

```diff
@@ -438,55 +445,65 @@ text-char = %x0-5B         ; omit \
           / %xE000-10FFFF
 ```
 
+### Literals
+
+_Literal_ is used for matching variants and providing input to _expressions_.
+_Quoted_ literals may include content with any Unicode code point,
+except for surrogate code points U+D800 through U+DFFF.
+The characters `\` and `|` MUST be escaped as `\\` and `\|`.
+_Unquoted_ literals have a much more restricted range that
+is intentionally close to the XML's [Nmtoken](https://www.w3.org/TR/xml/#NT-Nmtoken),
+with the restriction that it MUST NOT start with `-` or `:`,
+as those would conflict with _function_ start characters.
```

Comment:
These two lines don't feel _normative_ the way the rest of this passage does. Perhaps remove them?

Reply:
This isn't quite how I would approach this. I think it needs more explanation about literals in general and then about the unquoted ones. The relationship to Nmtoken (which we broke on purpose) isn't that relevant any more. Perhaps:

```suggestion
_Literal_ values are used to pass data to various parts of a `message`:
* As the value of a `key` in a `when` statement
* As the `argument` in an `expression`
* As the `value` in an `option`

A `Literal` is a sequence of _Unicode code points_ and can include any Unicode character. Surrogate code points are not allowed. 

The characters `\\` U+005C REVERSE SOLIDUS and `|` U+007C VERTICAL BAR **_must_** be escaped (as `\\` and `\|` respectively) when they appear in the value of a `Literal`.

Spaces are significant in a `Literal`.

A `Quoted` literal is surrounded by `|` characters. 

A `Literal` can be `Unquoted` when its content matches that production. The content restrictions for `Unquoted` follow best practices for the use of Unicode in formal grammars and are intentionally similar to, for example, XML's [Nmtoken](https://www.w3.org/TR/xml/#NT-Nmtoken).
```

Reply:
> +A `Literal` is a sequence of _Unicode code points_ and can include any Unicode character. Surrogate code points are not allowed. 

Should be:

A `Literal` is a sequence of _Unicode code points_, and can contain any Unicode code points except for surrogate code points and non-character code points.

Reason: "Unicode character" would mean "assigned Unicode character", which is unnecessarily fragile across versions.

Reply:
Actually, non-characters (U+FFFF for example) are not excluded. Only surrogate code points are. This is consistent with e.g. DOMString and USVString.

Reply:
That's fine